### ISSUE TITLE
time embedding use cat[cos, sin]

### ISSUE THREAD

**Issue by shy19960518** [facebookresearch/DiT#78]
Here is the time embedding code in the TimestepEmbedder() class in model.py ## at the top of Dit class
_embedding = torch.cat([torch.cos(args), torch.sin(args)], dim=-1)_

It seems to connect two tensors in the dimension of half: [cos0, cos1,..., cos(half), sin0, sin1, ..., sin(half)]
However, the formula is alternately inserted: [cos0, sin0, cos1, sin1, ..., cos(half), sin(half)]
Experiments showed that the model works well in A. But I wonder that will this have any impact? Or In what kind of task it will lead to hazard? 

Another question is why this part used an MLP for another representation for t? I mean, the representation/understanding could be processed in Transformer.
